sublime-music: 0.11.10 -> 0.11.11

sumnerevans opened the pull request:
This upgrade includes a switch from setuptools to poetry for the build. It also converts to using the test suite provided by the Sublime Music upstream project rather than just the import check.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

https://gitlab.com/sublime-music/sublime-music/-/releases/v0.11.11

###### Things done

```
$ nix-shell -p nixpkgs-review --run "nixpkgs-review rev ad6389e618987a29371cb5e506208400248868b9"
...
[nix-shell ...] $ sublime-music
```

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

sumnerevans committed:
sublime-music: 0.11.10 -> 0.11.11

This upgrade includes a switch from setuptools to poetry for the build

SuperSandro2000 commented on pkgs/applications/audio/sublime-music/default.nix:
```diff
@@ -19,7 +27,7 @@ python3Packages.buildPythonApplication rec {
   nativeBuildInputs = [
     gobject-introspection
-    python3Packages.setuptools
+    python3Packages.poetry
```
Can we use poetry-core?

sumnerevans replied:
I get this error:
```
ModuleNotFoundError: No module named 'poetry.masonry'
```
when I use `poetry-core` instead of `poetry`.

SuperSandro2000 replied:
You can fetchpatch https://gitlab.com/sublime-music/sublime-music/-/commit/9b0af19dbdfdcc5a0fa23e73bb34c7135a8c2855.patch from https://gitlab.com/sublime-music/sublime-music/-/merge_requests/60

SuperSandro2000 commented on pkgs/applications/audio/sublime-music/default.nix:
```diff
@@ -53,9 +61,12 @@ python3Packages.buildPythonApplication rec {
 
-  # no tests
-  doCheck = false;
-  pythonImportsCheck = [ "sublime_music" ];
```
Why did you remove the importsCheck?

sumnerevans replied:
I am now using the upstream unit tests directly, rather than relying on the imports check. https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck says that it's "highly preferred" to use unit tests over the `pythonImportsCheck`.

If you want I can add it back, but it's redundant considering that the unit tests import the module and much more.

SuperSandro2000 replied:
checkInput can contain extra packages which can hide import errors. It doesn't hurt to add a pythonImportsCheck and adds some safety to not miss depedencies.

sumnerevans commented:
@albakham would you be able to review this as well?

sumnerevans committed:
sublime-music: use poetry-core

Pulling in the patch from https://gitlab.com/sublime-music/sublime-music/-/merge_requests/60
to use poetry-core instead of poetry
